libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -96,40 +96,50 @@ func NewPodEventer(uuid uuid.UUID, cfg *conf.C, client k8s.Interface, publish fu
 		return nil, fmt.Errorf("couldn't create watcher for %T due to error %w", &kubernetes.Pod{}, err)
 	}
 
-	options := kubernetes.WatchOptions{
-		SyncTimeout: config.SyncPeriod,
-		Node:        config.Node,
```

Comment:
I don't think that `Node` and `Namespace` fields are related with the `add_resource_metadata` settings at all.
Removing them from here would mean that one cannot scope on `node` and `namespace` if `add_resource_metadata.node/namespace` is not enabled.

So I think this removal should reverted.

You can find more info in the docs about the `node` and `namespace` scope settings at https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-autodiscover.html#_kubernetes.

Reply:
They were changed in [this PR](https://github.com/elastic/beats/pull/22189) a long time ago.

> attach node metadata on pod metadata by default (without add_resource_metadata required to enabled) for both events coming from autodiscovered pods but also for events enriched by add_kubernetes_metadata

We need to find a way to stop these watchers, otherwise it is never optional like description suggests.

Reply:
I'm not referring to the Watchers but to the `Node` and `Namespace` fields of the `WatchOptions` struct. These are completely irrelevant to the `add_resource_metadata` settings.

So with the removal of the `Node` from the initialization of the `WatchOptions` the `node` scope setting at https://github.com/elastic/beats/blob/cc39376a5d4c87cc1a034396f8fe6f0fb54dec73/deploy/kubernetes/filebeat-kubernetes.yaml#L136 will have no impact when the `add_resource_metadata` is not enabled.

My point is that `WatchOptions` should include the `Node` and `Namespace` settings in nay case regardless of the `add_resource_metadata` settings. But with this PR, these fields are not defined inside the `if` blocks. This should not be the case, right? Or do I miss anything here?

Reply:
Sorry, I didn't understand at first. They are not defined indeed, but they were also not being used before

Reply:
Should I include the node and namespace for all 4 watchers (cronjob, deployment, namespace and node)?

Reply:
I think that the initialization of the `WatchOptions` should remain as it was before this patch.

In order to fix the issue you only need to surround https://github.com/elastic/beats/blob/8effe2400de59546c6243caa36f4ae6356ac6091/libbeat/autodiscover/providers/kubernetes/pod.go#L110-L115 and https://github.com/elastic/beats/blob/8effe2400de59546c6243caa36f4ae6356ac6091/libbeat/autodiscover/providers/kubernetes/pod.go#L106-L109 with the proper `if` blocks.

Makes sense?

Reply:
In GetPodMetaGen the metaconf settings are already checked and if disabled the watchers for node and namespace are omitted .
An easy solution to our case could also be to initialize the `p` omitting the namespace and node watcher and only set them if the relevant fields are enabled in metacond.

Reply:
Makes sense @ChrsMark , but notice that namespace watcher is not using those options, only node watcher

Reply:
Ah now I see why it was confusing @constanca-m. 
The `options` variable is only used by the Node's `Watcher` and nowhere else. Initially I was confused by seeing this being removed that's why I asked for the reasoning for removing this since it looked irrelevant to this PR's purpose.
Anyways, I think it is safe to create different `options` structs per Watcher like the following:
- In Node's watcher only use the `config.Node` scoping setting (namespace does not make sense for Node resources anyways)
- In Namespace's watcher only use the `config.Namespace` setting.

With this you can indeed remove completely the `options` variable initialization from line 99. Sorry for the confusion here.

Reply:
@MichaelKatsoulis unfortunately these Watchers are not only used for the `add_resource_metadata` part. These are also used for autodiscovery scenarios which is irrelevent to the `add_resource_metadata` setting. See my comment at https://github.com/elastic/beats/pull/37181/files#r1402587441.

Reply:
Thanks @ChrsMark for the clarification. I updated the options and it should be correct now.

Reply:
Looks good, thanks!